core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs.py

```diff
@@ -0,0 +1,123 @@
+
+MYPY = False
+if MYPY:  # pragma: no cover
+    from mypy_imports import suggestion_models
+
+(suggestion_models, ) = models.Registry.import_models([
+    models.Names.SUGGESTION])
+
+
+# TODO(#15613): Here we use MyPy ignore because the incomplete typing of
+# apache_beam library and absences of stubs in Typeshed, forces MyPy to
+# assume that PTransform class is of type Any. Thus to avoid MyPy's error
+# (Class cannot subclass 'PTransform' (has type 'Any')), we added an
+# ignore here.
+class GetDeprecatedSuggestionEditStateContentModels(beam.PTransform): # type: ignore[misc]
+    """Transform that gets all edit state content suggestion models."""
+
+    def expand(
+        self, pipeline: beam.Pipeline
+    ) -> Tuple[
+        beam.PCollection[suggestion_models.GeneralSuggestionModel],
+        beam.PCollection[job_run_result.JobRunResult]
+    ]:
+        suggestion_edit_state_content_model_to_delete = (
+            pipeline
+            | 'Get all general suggestion models' >> ndb_io.GetModels(
+                suggestion_models.GeneralSuggestionModel.get_all())
+            | 'Filter edit state content suggestion' >> (
+                beam.Filter(
+                    lambda model: ((
+                        model.suggestion_type == (
+                            feconf.SUGGESTION_TYPE_EDIT_STATE_CONTENT)) and (
+                                model.target_type == (
+                                    feconf.ENTITY_TYPE_EXPLORATION)))
+                ))
+        )
+
+        suggestion_edit_state_content_model_to_delete_count = (
+            suggestion_edit_state_content_model_to_delete
+            | 'Count edit state content suggestion to be deleted' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'EDIT STATE CONTENT SUGGESTION'))
+        )
+
+        return (
+                suggestion_edit_state_content_model_to_delete,
+                suggestion_edit_state_content_model_to_delete_count,
+        )
+
+
+class DeleteDeprecatedSuggestionEditStateContentModelsJob(base_jobs.JobBase):
+    """Job that deletes edit state content suggestion models as these are
+    deprecated.
+    """
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+
+        (suggestion_edit_state_content_model_to_delete, (
+        suggestion_edit_state_content_model_to_delete_result)) = (
+            self.pipeline
+            | 'Get edit state content suggestion models' >> (
+                GetDeprecatedSuggestionEditStateContentModels())
+        )
+
+        unused_models_deletion = (
+            (
+                suggestion_edit_state_content_model_to_delete
+            )
+            | 'Extract keys' >> beam.Map(lambda model: model.key)
+            | 'Delete models' >> ndb_io.DeleteModels()
+        )
+
+        return (
+            (
+                suggestion_edit_state_content_model_to_delete_result,
+            )
+            | 'Merge results' >> beam.Flatten()
```

Comment:
Just wondering, why do we not report these the same way as in the audit job?

Reply:
Totally agree, we actually don't need to use beam.Flatten() here.
Made required changes.